php/libraries/User.class.inc

```diff
@@ -92,70 +94,96 @@ class User extends UserPermissions implements
 
-        $user_pid = $DB->pselectCol(
-            "SELECT ProjectID FROM user_project_rel upr WHERE upr.UserId=:uid",
-            ['uid' => $row['ID']]
+        // ------------------- Extract User Projects -------------------
+
+        $user_projects_query = $DB->pselect(
+            "
+             SELECT upr.ProjectID, p.Name
+             FROM user_project_rel upr
+             LEFT JOIN Project p ON (upr.ProjectID = p.ProjectID)
+             WHERE upr.UserID = :UID
+            ",
+            ['UID' => $row['ID']]
         );
-        foreach ($user_pid as $k=>$pid) {
-            $user_pid[$k] = ProjectID::singleton(intval($pid));
+
+        $user_pid      = [];        // Array to store ProjectID objects
+        $project_names = [];   // Array to store project names
+
+        foreach ($user_projects_query as $key => $val) {
+            // Convert ProjectID to ProjectID object, ensuring it's a string
+            $user_pid[$key]  = \ProjectID::singleton($val['ProjectID']);
+            $project_names[] = $val['Name'];
         }
 
-        // Get examiner information
+        // Assign ProjectIDs and concatenated project names to userInfo
+        $row['ProjectIDs'] = $user_pid;
+        $row['Projects']   = implode(';', $project_names);
+
+        // ------------------- Extract Examiner Information -------------------
         $examiner_check = $DB->pselect(
             "SELECT e.full_name,
                     epr.centerID,
                     e.radiologist,
                     epr.active,
                     epr.pending_approval
-                  FROM examiners e
-                  JOIN examiners_psc_rel epr ON (e.examinerID=epr.examinerID)
-                  WHERE e.userID=:uid
-                    AND (epr.active='Y'
-                          OR (epr.active='N' AND epr.pending_approval='Y')
-                        )",
+              FROM examiners e
+              JOIN examiners_psc_rel epr ON (e.examinerID = epr.examinerID)
+              WHERE e.userID = :uid
+                AND (epr.active = 'Y'
+                      OR (epr.active = 'N' AND epr.pending_approval = 'Y'))
+            ",
             [
                 "uid" => $row['ID'],
             ]
         );
 
-        $examiner_info =[];
-        if (!empty($examiner_check) && !is_null($examiner_check)) {
+        $examiner_info = [];
+
+        if (!empty($examiner_check)&& !is_null($examiner_check)) {
             foreach ($examiner_check as $val) {
                 if ($val['active'] == 'Y') {
+                    // Store pending approval status if examiner is active
                     $examiner_info['pending'] = $val['pending_approval'];
                 }
-                $examiner_info[$val['centerID']] =[
+                /*
+                Store examiner's centerID with active status and radiologist flag,
+                ensuring centerID is a string
+                */
+                $examiner_info[(string)$val['centerID']] = [
```

Comment:
Do we need that change? Does it solve an error?

Reply:
it was to stay consistent with my changes when I used to cast the projects and sites into strings but it doesn't affect functionality so it can removed.